sunpy/util/tests/test_config.py

```diff
@@ -80,3 +86,31 @@ def test_get_and_create_sample_dir():
     # Set the config back
     os.rmdir(new_path)
     config.set('downloads', 'sample_dir', os.path.join(USER, 'sunpy', 'data', 'sample_data'))
+
+
+def test_write_config(tmpdir, undo_config_dir_patch):
+    config_filename = 'sunpyrc'
+
+    # Try to set a manual one (already created)
+    tmp_config_dir = tmpdir.mkdir("config_here")
+    os.environ["SUNPY_CONFIGDIR"] = tmp_config_dir.strpath
+
+    assert tmp_config_dir == _get_user_configdir()
+
+    config_file = Path(sunpy.__file__).parent / 'data' / config_filename
+    user_config_file = tmp_config_dir / config_filename
+
+    tmp_file = tmpdir.join("hello.txt")
+    # Have to write to the file otherwise its seen as a directory(?!)
+    tmp_file.write("content")
+
+    write_config()
+    assert open(user_config_file, 'r').read() == open(config_file, 'r').read()
+
+    with pytest.warns(SunpyUserWarning):
+        write_config(overwrite=False)
```

Comment:
Isn't `overwrite=False` the default behaviour? I would not add it here if that's what we expect by default. Otherwise if someone changes the default we won't notice the test failing.